src/benchmark_runner.cc

```diff
@@ -236,8 +242,11 @@ class BenchmarkRunner {
     VLOG(2) << "Ran in " << i.results.cpu_time_used << "/"
             << i.results.real_time_used << "\n";
 
+    // By using KeepRunningBatch a benchmark can iterate more times than
+    // requested, so take the iteration count from i.results.
+    i.iters = i.results.iterations_per_thread;
```

Comment:
So this is just `i.iters = i.results.iterations / b.threads;` ?

Reply:
i think that's right :)

Reply:
I agree.  Shall I change this line to `i.iters = i.results.iterations / b.threads;`?

That would simplify the patch and avoid the concern that the CHECK would trip.  I could even be more conservative and use `i.iters = std::max(iters, i.results.iterations / b.threads);`

Reply:
Let's go for division, i guess.

Reply:
Sounds good, done.